app/scripts/controllers/service.js

```diff
@@ -60,6 +99,36 @@ angular.module('openshiftConsole')
           }
         );
 
+        watches.push(DataService.watch("services", context, function(services) {
+          $scope.services = services.by("metadata.name");
+        }));
+
+        watches.push(DataService.watch("pods", context, function(pods) {
+          $scope.pods = [];
```

Comment:
there shouldn't be any reason for converting the pod map to an array, you also probably have no reason to have all pods on the scope, you only care about the pods for the service from within the view.  You can use a local variable to the controller to keep track of the full set of pods.